packages/dds/tree/src/test/feature-libraries/optional-field/optionalFieldUtils.ts

```diff
@@ -4,17 +4,119 @@
  */
 
 import { strict as assert } from "assert";
-import { TaggedChange } from "../../../core/index.js";
-// eslint-disable-next-line import/no-internal-modules
-import { RegisterId, OptionalChangeset } from "../../../feature-libraries/optional-field/index.js";
+import { ChangesetLocalId, TaggedChange, makeAnonChange } from "../../../core/index.js";
+import {
+	RegisterId,
+	OptionalChangeset,
+	Move,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/optional-field/index.js";
+import { Mutable } from "../../../util/index.js";
+
+function asRegister(input: RegisterId | ChangesetLocalId): RegisterId {
+	if (typeof input === "string" || typeof input === "object") {
+		return input;
+	}
+	return { localId: input };
+}
+
+export const Change = {
+	/**
+	 * @returns An empty changeset
+	 */
+	empty: (): OptionalChangeset<never> => ({ moves: [], childChanges: [] }),
+	/**
+	 * @param src - The register to move a node from. The register must be full in the input context of the changeset.
+	 * @param dst - The register to move that node to.
+	 * The register must be empty in the input context of the changeset, or emptied as part of the changeset.
+	 * @returns A changeset that moves a node from src to dst.
+	 */
+	move: (
+		src: RegisterId | ChangesetLocalId,
+		dst: RegisterId | ChangesetLocalId,
+	): OptionalChangeset<never> => ({
+		moves: [[asRegister(src), asRegister(dst), "nodeTargeting"]],
+		childChanges: [],
+	}),
+	/**
+	 * @param target - The register remove a node from. The register must be full in the input context of the changeset.
+	 * @param dst - The register to move the contents of the target register to.
+	 * The register must be empty in the input context of the changeset, or emptied as part of the changeset.
+	 * @returns A changeset that clears a register and moves the contents to another register.
+	 */
+	clear: (
+		target: RegisterId | ChangesetLocalId,
+		dst: RegisterId | ChangesetLocalId,
+	): OptionalChangeset<never> => ({
+		moves: [[asRegister(target), asRegister(dst), "cellTargeting"]],
+		childChanges: [],
+	}),
+	/**
+	 * @param target - The register to reserve. The register must NOT be full in the input context of the changeset.
+	 * @param dst - The register that the contents of the target register should be moved to should it become populated.
+	 * The register must be empty in the input context of the changeset, or emptied as part of the changeset.
+	 * @returns A changeset that reserves an register.
+	 */
+	reserve: (
+		target: RegisterId | ChangesetLocalId,
+		dst: RegisterId | ChangesetLocalId,
+	): OptionalChangeset<never> => {
+		assert(target === "self", "Reserve cell only supports self as source");
```

Comment:
Why does this function take a target if it only has one legal value?

Reply:
Because the format I want to move to supports this concept for any register.